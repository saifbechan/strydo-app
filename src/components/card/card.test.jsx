import React from 'react';
import { render } from '@testing-library/react';
import { useDrag, useDrop } from 'react-dnd';
import 'jest-styled-components';

import Card from './card.component';

jest.mock('react-dnd');

useDrag.mockImplementation(() => [{ isDragging: false }, () => {}]);
useDrop.mockImplementation(() => [{ isOver: false }, () => {}]);

const children = 'child';

it('renders the first card correctly', () => {
  const props = {
    id: 'abc-123',
    idx: 0
  };

  const { asFragment } = render(<Card {...props}>{children}</Card>);
  expect(asFragment()).toMatchSnapshot();
});

it('renders the second and further cards correctly', () => {
  const props = {
    id: 'abc-123',
    idx: 1
  };

  const { asFragment } = render(<Card {...props}>{children}</Card>);
  expect(asFragment()).toMatchSnapshot();
});
