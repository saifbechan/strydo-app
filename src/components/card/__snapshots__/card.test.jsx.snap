// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  border: 1px solid silver;
  cursor: pointer;
  margin: 5px;
  padding: 15px 20px;
}

.c0:hover {
  background-color: whitesmoke;
}

.c1 {
  cursor: pointer;
  margin-top: 15px;
  padding: 5px;
  text-align: center;
  width: 100%;
}

.c1:hover {
  color: grey;
}

<div
    class="c0"
    data-testid="card"
    id="abc-123"
  >
    child
    <button
      class="c1"
      data-testid="remove-card-handler"
    >
      - remove card
    </button>
  </div>
</DocumentFragment>
`;
